geowrangler/datasets/utils.py

```diff
@@ -0,0 +1,78 @@
+# AUTOGENERATED! DO NOT EDIT! File to edit: notebooks/13_datasets_utils.ipynb (unless otherwise specified).
+
+__all__ = ["urlretrieve", "make_report_hook"]
+
+
+# Internal Cell
+
+import contextlib
+import gzip
+import hashlib
+import json
+import os
+import shutil
+from pathlib import Path
+from types import SimpleNamespace
+from urllib.error import ContentTooShortError, HTTPError
+from urllib.parse import urlparse
+
+import numpy as np
+import requests
+from fastcore.net import urldest, urlopen
+from fastprogress.fastprogress import progress_bar
+from loguru import logger
+from shapely.geometry import box
+
+import geowrangler.raster_process as rp
+import geowrangler.raster_zonal_stats as rzs
+
+# Cell
+# from https://github.com/fastai/fastcore/blob/86337bad16a65f23c5335286ab73cd4d6425c586/fastcore/net.py#L147
+# add headers to urlwrap call (to allow auth)
+def urlretrieve(
+    url, filename, headers=None, reporthook=None, timeout=None, chunksize=8192
+):
+    "Same as `urllib.request.urlretrieve` but also works with `Request` objects"
```

Comment:
see [urlretrieve might be deprecated discussion](https://stackoverflow.com/questions/15035123/what-command-to-use-instead-of-urllib-request-urlretrieve/15035466#15035466)

This is just a method name I chose because its where I copied it from (doesn't have to be named urlretrieve) to incorporate a reporthook  to allow the addition of a progress bar.

The reason I  replaced `shutil.copyfileobj` is that doesn't have a callback hook to track the progress of the copy.
Anecdata seems to indicate a better ux and no performance degradation.